Improve create playlist UX

PikachuEXE opened the pull request:
## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue
<!-- Please link the issue your pull request is referring to. -->
<!-- If this pull request fully resolves the relevant issue, put "closes" before the issue number. -->
<!-- Example: "closes #123456". -->

## Description
<!-- Please write a clear and concise description of what the pull request does. -->
- Make it impossible to create playlists with spaces wrapped name
- Show duplicate playlist message sooner (when name input instead of clicking create button)
- Update default new playlist name to include channel name when copying a remote playlist

## Screenshots <!-- If appropriate -->
<!-- Please add before and after screenshots if there is a visible change. -->
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/dd6529bd-60b2-45d7-8c76-8f8b9fe5a10e)
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/c584617a-ccf2-4cde-ac5f-af6f76a6c401)
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/848a1fb9-6104-45ca-85ab-8aacf8b1e4b7)


## Testing <!-- for code that is not small enough to be easily understandable -->
<!-- Has this pull request been tested? -->
<!-- Please describe shortly how you tested it. -->
<!-- Are there any ramifications remaining? -->
- Input spaces around playlist name inside create playlist prompt
- ensure created playlist got name trimmed
- ensure cannot create playlist if another playlist with same trimmed name exists (and got message shown)

- Copy local playlist, create playlist - unchanged
- Copy remote playlist, create playlist - ensure channel name included

## Desktop
<!-- Please complete the following information-->
- **OS:**
- **OS Version:**
- **FreeTube version:**

## Additional context
<!-- Add any other context about the pull request here. -->

jasonhenriquez reviewed (commented):
**suggestion (non-blocking):** Could you give the same treatment for the empty playlist name case?

PikachuEXE commented:
> Could you give the same treatment for the empty playlist name case?

Just did & updated OP

jasonhenriquez reviewed (approved):
Thanks Pika. It's a tiny bit jarring having the empty playlist alert showing from the jump now, but maybe can switch to using  the `@disabled-click` + toast route for our `ft-button`s. For now, this looks good.

absidue commented on src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue:
```diff
@@ -15,13 +15,24 @@
         :maxlength="255"
         class="playlistNameInput"
-        @input="(input) => playlistName = input"
+        @input="(input) => handlePlaylistNameInput(input)"
```
```suggestion
        @input="handlePlaylistNameInput"
```

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Thanks Pika. It's a tiny bit jarring having the empty playlist alert showing from the jump now

Yep i agree. `Playlist name cannot be empty. Please input a name.` gives me as the user the impression that i did something wrong here but i didnt even had the chance to input something and its already telling me it cant be empty.

I suggest (maybe this isnt a good suggestion idk) that the first time the users gets to see that message is when they input something but remove the input. This way the user doesnt see it right away but does get to see it if they intentionally removed it

PikachuEXE commented:
Updated to only show empty playlist name when space input
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/565a904c-4027-4665-8a3f-166dd1c4943a)
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/8e3bbf38-fe80-4a9e-8204-3c1895483189)

> I like this very much but im not sure how this is relevant when copying random user playlists.

Not relevant, it should only work on remote playlist (with channel name present)

> Should also probably happen here

I already got `:value="playlistName"` on `ft-input` I got no idea why it won't auto update

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Not relevant, it should only work on remote playlist (with channel name present)

I dont mean Local created playlist. I mean a random user created a playlist of your favorite creator.

PikachuEXE commented:
> a random YT user created a playlist of your favorite creator

I see, but in that case it's still from that YT user not from the creator(s)
Which is why that's the default value but allows editing
Also what happens when the playlist is for 2+ creators? It's really up to the user to decide what's best for them
I have no idea how to programmatically determine for different users on different playlists about whether the channel name is relevant or not...

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Its indeed not that big of an issue but i thought i'd bring it up in case somebody thinks otherwise

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue:
```diff
@@ -15,13 +15,24 @@
     </ft-flex-box>
+    <ft-flex-box v-if="playlistNameBlank">
+      <p>
+        {{ $t('User Playlists.SinglePlaylistView.Toast["Playlist name cannot be empty. Please input a name."]') }}
```
Is this unused now? Im not getting this anymore

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Should also probably happen here

> > I already got :value="playlistName" on ft-input I got no idea why it won't auto update

Everything LGTM aside from this behaving weird.

PikachuEXE commented:
> Everything LGTM aside from this behaving weird.

Just fixed

efb4f5ff-1298-471a-8973-3d47447115dc commented:
spacebar isnt working for me anymore

PikachuEXE commented:
Reverted, no idea what `Should also probably happen here` means
Oh wait you mean the edit playlist info place? OOF

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Oh wait you mean the edit playlist info place

yes thats correct, sorry for being vague i thought the screenrec was enough

PikachuEXE commented:
- Fixed create button not disabled when input name blank (spaces only but not empty)
- Fixed playlist page showing text in edit mode
- Updated playlist page edit mode to show messages for empty/blank/duplicate input playlist name

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Fixed playlist page showing text in edit mode

Im not sure what you mean by this and I think I wasnt clear enough about what change I would like to see in edit mode.

What I would like to see is the following, trim spaces in edit mode like how its been done in the toast message, the playlist page and inside the playlist


https://github.com/FreeTubeApp/FreeTube/assets/73130443/add2ad05-9af0-4d9e-9e47-061a9f1620e9

PikachuEXE commented:
> Fixed playlist page showing text in edit mode

Oh forgot to include screenshot, before:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/30b6c70e-abb1-4da0-9c9a-e2c55be94a33)
after:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/129cdbb1-0253-4ef8-88f5-a2b6ca5a29e0)

> What I would like to see is the following

Yes I understand please test the latest version updated recently (though let me know if I miss anything coz create/edit playlist got some logic slightly different)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Retested, still seeing the spaces here

https://github.com/FreeTubeApp/FreeTube/assets/73130443/7e9121cd-d496-4fc9-8ae7-c920036a09a6

PikachuEXE commented:
I don't quite understand what you expect

If you are referring to what's displayed and the actual value (with 2+ spaces inside) that seems to be existing behaviour (not saying it's not a bug but not trying to handle it in this PR)
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/b535558a-57f7-4f48-b09a-a90ad5d82064)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Ah you're correct this is existing behavior. Im so confused rn but ill re-review all the changes for the sake of my sanity

PikachuEXE commented:
Sorry I totally forgot enter can do so many things
Also `@click` isn't the best event name to be linked with pressing enter
Fixed

jasonhenriquez commented on src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue:
```diff
@@ -15,13 +15,24 @@
     <ft-flex-box>
       <ft-button
         :label="$t('User Playlists.CreatePlaylistPrompt.Create')"
+        :disabled="playlistPersistenceDisabled"
```
**thought/question:** I have no idea how Vue knows to make the underlying button disabled when there doesn't seem to be a prop for this in our `ft-button` class, but it does.

absidue replied:
Vue calls that feature "fallthrough attributes", if the component doesn't have a prop for the attribute, it adds the attribute to the root/top-level HTML element inside that component.

Vue 3 has an entire docs page on it, but it also describes Vue 3 specific behaviour so it's not that useful for us yet, Vue 2's docs only have an offhand mention of it (https://v2.vuejs.org/v2/api/#inheritAttrs).

jasonhenriquez commented on src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.js:
```diff
@@ -33,26 +33,45 @@ export default defineComponent({
       return this.$store.getters.getNewPlaylistVideoObject
     },
+
+    playlistNameEmpty() {
```
**note:** once the Vue 3 migration is complete, this will be a great place for composable functions.
